clis/teliod/src/daemon.rs

```diff
@@ -47,32 +65,112 @@ fn telio_task(
         None,
     )?;
 
+    // Keep track of current meshnet states
+    let mut state = TelioTaskStates::default();
+    // Mock the interface configuration for tests
+    let sys_config: Box<dyn SystemConfigurator> = if auth_token.as_str().eq(EMPTY_TOKEN) {
+        // And some fake IP address
+        state.interface_ip_address = Some(IpAddr::V4(Ipv4Addr::new(100, 100, 100, 100)));
+        Box::new(MockSystemConfigurator::new(false))
+    } else {
+        Box::new(IfconfigSystemConfigurator)
+    };
+
     // TODO: This is temporary to be removed later on when we have proper integration
     // tests with core API. This is to not look for tokens in a test environment
     // right now as the values are dummy and program will not run as it expects
     // real tokens.
     if !auth_token.as_str().eq("") {
         start_telio(&mut telio, node_identity.private_key, interface_name)?;
-        task_retrieve_meshmap(node_identity, auth_token, tx_channel);
+
+        // Auto enable meshnet for the first time
+        if auto_start_meshnet {
+            task_enable_meshnet(tx_channel.clone());
+        }
 
         while let Some(cmd) = rx_channel.blocking_recv() {
             info!("Got command {:?}", cmd);
             match cmd {
                 TelioTaskCmd::UpdateMeshmap(map) => {
-                    if let Err(e) = telio.set_config(&Some(map)) {
-                        error!("Unable to set meshmap due to {e}");
+                    let new_ip_address = map
+                        .ip_addresses
+                        .as_deref()
+                        .and_then(|addresses| addresses.first().cloned());
+                    let address_changed = state.interface_ip_address != new_ip_address;
+                    state.interface_ip_address = new_ip_address;
+
+                    if state.meshnet_is_enabled {
+                        match telio.set_config(&Some(map)) {
+                            Ok(_) => {
+                                // configure the interface ip address
```

Comment:
would be nice to have also a log when meshmap is updated successfully.

Reply:
Added log